### DISCUSSION TITLE
Migrate `mist-cli` formula to cask?

### DISCUSSION THREAD

**Discussion by ninxsoft** [Homebrew/discussions#4833] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: e57a87cfe9c15f192cd4fc6695a2b3b7acf0ddf7
Last commit: 52 minutes ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 510fdb45d83607fc9b171a1df7f3d5552d77939e
Core tap last commit: 25 minutes ago
Core tap branch: master
Core tap JSON: 09 Oct 12:53 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_FROM_API: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode-15.0.0.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: 15.0 => /Applications/Xcode-15.0.0.app/Contents/Developer
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I am the maintainer of the following Mist tools, both currently supported by Homebrew:

| **Tool** | **Type** | **Formula** | **Cask** |
| :-: | :-: | :-: | :-: |
| [mist-cli](https://github.com/ninxsoft/mist-cli) | CLI | [mist-cli.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/m/mist-cli.rb) | - |
| [Mist.app](https://github.com/ninxsoft/Mist) | GUI | - | [mist.rb](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/m/mist.rb) |

I provide pre-compiled, [codesigned](https://developer.apple.com/support/code-signing/) and [notarized](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution) installer packages that can be downloaded via their respective GitHub repositories, preventing these sorts of [Gatekeeper](https://support.apple.com/en-us/HT202491) warnings to show when the apps are launched:

![macos-big-sur-alert-unverified-developer](https://github.com/Homebrew/discussions/assets/4812595/8473209e-4401-41d0-bb68-73968a5d7ddd)

For the **Mist.app** cask, this works great, as I point to the notarized package and Homebrew just makes it work ✅

For the **mist-cli** formula, my understanding is that bottles are compiled for each new version / architecture / major OS. The current drawback to this is that the `mist-cli` CLI binary loses (or never gains) any codesigning + notarization as originally intended. You can see this by verifying the code signature:

```bash
spctl --assess -vvv --type install /opt/homebrew/Cellar/mist-cli/2.0/bin/mist 

/opt/homebrew/Cellar/mist-cli/2.0/bin/mist: rejected
```

Versus a binary installed via the non-Homebrew method:

```bash
spctl --assess -vvv --type install /usr/local/bin/mist

/usr/local/bin/mist: accepted
source=Notarized Developer ID
origin=Developer ID Application: Nindi Gill (7K3HVCLV7Z)
```

My hope is to update the **mist-cli** formula to install using the pre-built packages provided in the GitHub repository. At first glance, switching to a cask sounds like the simplest solution, but I am not 100% sure as I came across this bullet point under [Rejected Casks](https://docs.brew.sh/Acceptable-Casks#rejected-casks):

> App is both open-source and CLI-only (i.e. it only uses the binary artifact). In that case, and [in the spirit of deduplication](https://github.com/Homebrew/homebrew-cask/issues/15603), submit it first to [homebrew/core](https://github.com/Homebrew/homebrew-core) as a formula that builds from source. If it is rejected, you may then try again as a cask (link to the issue from your pull request so we can see the discussion and reasoning for rejection).

Hoping to get some guidance / assistance on the following:

- Can a formula be configured to install CLI tools via installer packages (.pkg)?
- Is **mist-cli** a suitable candidate for submission to create a new cask / deprecate the old formula?
- Am I just holding it wrong / is there a simple solution or example you can point me towards?

Unfortunately I was not able to find a definitive path forward during my research - thank you in-advance for your time 🙏

### Relevant casks

mist

**Comment by SMillerDev** [comment]
> Can a formula be configured to install CLI tools via installer packages (.pkg)?

No, since formulae are source builds by definition.

> Is mist-cli a suitable candidate for submission to create a new cask / deprecate the old formula?

Seeing how the only difference would be the signing status, I don't think it is.

**Reply by ninxsoft** [reply]
Thanks for the response 🙌 I can inform users of the minimum (Xcode) requirements to build mist-cli from source
